test/unit/const_string.cpp

```diff
@@ -132,6 +159,22 @@ struct const_string_test
                 BOOST_TEST(s == "abc");
                 BOOST_TEST(! s.empty());
             }
+            {
+                const_string s;
+                {
+                    factory a;
+                    s = a(10,
```

Comment:
You need to expose `sizeof(base*)` as a public constexpr constant so that tests can be written which guarantee going into small or dynamic storage, even if the size of the small buffer is changed.